src/psyclone/psyir/symbols/datasymbol.py

```diff
@@ -0,0 +1,539 @@
+                self.copy_properties(extern_symbol)
+                self.interface = tmp
+            else:
+                raise NotImplementedError(
+                    "Error trying to resolve symbol '{0}' properties, the lazy"
+                    " evaluation of '{1}' interfaces is not supported."
+                    "".format(self.name, self.interface))
+
+    @property
+    def datatype(self):
+        '''
+        :returns: datatype of the DataSymbol.
+        :rtype: str
+        '''
+        return self._datatype
+
+    @datatype.setter
+    def datatype(self, value):
+        ''' Setter for DataSymbol datatype.
+
+        :param str value: new value for datatype.
+
+        :raises TypeError: if value is not a str.
+        :raises NotImplementedError: if the specified data type is invalid.
+        '''
+        if not isinstance(value, str):
+            raise TypeError(
+                "The datatype of a DataSymbol must be specified using a str "
+                "but got: '{0}'".format(type(value).__name__))
+        if value not in DataSymbol.valid_data_types:
+            raise NotImplementedError(
+                "DataSymbol can only be initialised with {0} datatypes but "
+                "found '{1}'.".format(str(DataSymbol.valid_data_types), value))
+        self._datatype = value
+
+    @property
+    def shape(self):
+        '''
+        :returns: shape of the symbol in column-major order (leftmost \
+                  index is contiguous in memory). Each entry represents \
+                  an array dimension. If it is 'None' the extent of that \
+                  dimension is unknown, otherwise it holds an integer \
+                  literal or a reference to an integer symbol with the \
+                  extent. If it is an empty list then the symbol \
+                  represents a scalar.
+        :rtype: list
+        '''
+        return self._shape
+
+    @property
+    def interface(self):
+        '''
+        :returns: the an object describing the interface to this DataSymbol.
+        :rtype: Sub-class of \
+            :py:class:`psyclone.psyir.symbols.datasymbol.DataSymbolInterface`
+        '''
+        return self._interface
+
+    @interface.setter
+    def interface(self, value):
+        '''
+        Setter for the Interface associated with this DataSymbol.
+
+        :param value: an Interface object describing how the DataSymbol is \
+                      accessed by the code.
+        :type value: Sub-class of \
+            :py:class:`psyclone.psyir.symbols.datasymbol.DataSymbolInterface`
+
+        :raises TypeError: if the supplied `value` is of the wrong type.
+        '''
+        if not isinstance(value, DataSymbolInterface):
+            raise TypeError("The interface to a DataSymbol must be a "
+                            "DataSymbolInterface but got '{0}'".
+                            format(type(value)))
+        self._interface = value
```

Comment:
pylint complains that `_interface` is not defined in the constructor.

Reply:
And it is right, added definition in the constructor